bishop.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "enums.h"
+#include "piece.h"
```

Comment:
Forward declare the class `Piece` to optimize compilation.

```suggestion
class Piece;
```

Reply:
It cannot be forward declared since it is used as a base class and the compiler needs a complete type.